compiler/src/compile.rs

```diff
@@ -383,13 +385,26 @@ impl<O: OutputStream> Compiler<O> {
                 items,
                 body,
             } => {
-                if *is_async {
-                    unimplemented!("async with");
-                } else {
-                    let end_label = self.new_label();
-                    for item in items {
+                let is_async = *is_async;
+
+                let end_labels = items
+                    .iter()
+                    .map(|item| {
```

Comment:
I'm still not a great fan of inline lambda functions like this. I would prefer a plain for loop and a mutable `end_labels` variable in this case.